server.go

```diff
@@ -0,0 +1,159 @@
+package dhstore
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"strings"
+
+	"github.com/mr-tron/base58"
+	"github.com/multiformats/go-multihash"
+)
+
+type server struct {
+	s   *http.Server
+	dhs DHStore
+}
+
+func newHttpServer(dhs DHStore, addr string) (*server, error) {
+	var dhss server
+	mux := http.NewServeMux()
+	mux.HandleFunc("/multihash", dhss.handleMh)
+	mux.HandleFunc("/metadata", dhss.handleMetadata)
+	dhss.s = &http.Server{
+		Addr:    addr,
+		Handler: mux,
+	}
+	dhss.dhs = dhs
+	return &dhss, nil
+}
+
+func (s *server) handleMh(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodPut:
+		s.handlePutMh(w, r)
+	case http.MethodGet:
+		s.handleGetMh(w, r)
+	default:
+		_, _ = io.Copy(io.Discard, r.Body)
+		_ = r.Body.Close()
+		http.Error(w, "", http.StatusNotFound)
+	}
+}
+
+func (s *server) handlePutMh(w http.ResponseWriter, r *http.Request) {
+	var mir MergeIndexRequest
+	err := json.NewDecoder(r.Body).Decode(&mir)
+	_ = r.Body.Close()
+	if err != nil {
+		http.Error(w, "", http.StatusBadRequest)
+		return
+	}
+
+	// TODO: Use pebble batch which will require interface changes.
+	//       But no big deal for now because every write to pebble is by NoSync.
+	for _, merge := range mir.Merges {
+		if err := s.dhs.MergeIndex(merge.Key, merge.Value); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
```

Comment:
Consider doing this in a separate goroutine so that the API can return right away without having to wait for the data to be processed, and possibly having the client time out. There may be a large batch to process.

Reply:
Thank you; will address in a separate PR with worker and backoff management. Captured https://github.com/ipni/dhstore/issues/5